compiler/qsc/src/bin/qsi.rs

```diff
@@ -81,54 +85,91 @@ impl Receiver for TerminalReceiver {
     }
 }
 
+#[allow(clippy::too_many_lines)]
 fn main() -> miette::Result<ExitCode> {
     let cli = Cli::parse();
-    let mut sources = cli
-        .sources
-        .iter()
-        .map(read_source)
-        .collect::<miette::Result<Vec<_>>>()?;
-
     let mut features = LanguageFeatures::from_iter(cli.features);
 
-    if sources.is_empty() {
-        let fs = StdFs;
-        if let Some(qsharp_json) = cli.qsharp_json {
-            if let Some(dir) = qsharp_json.parent() {
-                let project = match fs.load_project(dir, None) {
-                    Ok(project) => project,
-                    Err(errs) => {
-                        for e in errs {
-                            eprintln!("{e:?}");
-                        }
-                        return Ok(ExitCode::FAILURE);
+    let (store, dependencies, source_map) = if let Some(qsharp_json) = cli.qsharp_json {
```

Comment:
Similarly to my other comment on `qsc`, it would be worth breaking down the logic here into multiple fucntions such that it's easier to see where the values of `store`, `dependencies` and `source_map` come from.